cmd/webhook/webhook.go

```diff
@@ -321,7 +338,15 @@ func mutatePods(ar v1beta1.AdmissionReview) *v1beta1.AdmissionResponse {
 	}
 	// Add volumes if any container was patched for pinning
 	if pinningPatchAdded {
-		patchList = patchVolumesForPinning(patchList)
+		alreadyPatched := false
+		for _, volume := range pod.Spec.Volumes{
```

Comment:
Would it be enough to check only this in the beginning of this function and remove all the other checks ?

Reply:
You're right! I moved a check in the begining.